fearless/Modules/Staking/Analytics/Rewards/AnalyticsRewardsPresenter.swift

```diff
@@ -0,0 +1,105 @@
+import Foundation
+import BigInt
+
+final class AnalyticsRewardsPresenter {
+    weak var view: AnalyticsRewardsViewProtocol?
+    let wireframe: AnalyticsRewardsWireframeProtocol
+    let interactor: AnalyticsRewardsInteractorInputProtocol
+    private let viewModelFactory: AnalyticsRewardsViewModelFactoryProtocol
+    private var rewardsData = [SubqueryRewardItemData]()
+    private var selectedPeriod = AnalyticsPeriod.default
+    private var selectedPeriodDiff = 0
+    private var priceData: PriceData?
+    private var stashItem: StashItem?
+
+    init(
+        interactor: AnalyticsRewardsInteractorInputProtocol,
+        wireframe: AnalyticsRewardsWireframeProtocol,
+        viewModelFactory: AnalyticsRewardsViewModelFactoryProtocol
+    ) {
+        self.interactor = interactor
+        self.wireframe = wireframe
+        self.viewModelFactory = viewModelFactory
+    }
+
+    private func updateView() {
+        let viewModel = viewModelFactory.createViewModel(
```

Comment:
Consider to store calendar in presenter and pass to view model factory. Also we need to subscribe to day changes using this calendar and update view.